source-azure-blob-storage/schema.json

```diff
@@ -0,0 +1,69 @@
+{
```

Comment:
I actually like using `go:embed` for this quite a bit, but for consistency I think we should just have the schema definition as a in-line string like we do for [others](https://github.com/estuary/connectors/blob/cd7c1b12ed2aa450e24953e2d5df18143f12e4ad/source-s3/main.go#L279-L348)

_Really_ what we should do is use our [schema-gen](https://github.com/estuary/connectors/tree/main/go/schema-gen) to generate the schema from the config struct tags (see [example](https://github.com/estuary/connectors/blob/cd7c1b12ed2aa450e24953e2d5df18143f12e4ad/source-dynamodb/main.go#L16-L28)), but schema-gen doesn't support things like `oneOf` configurations for auth methods so we haven't done that generally. But that is the eventual goal, and we could then remove all of the hand-written schemas, so best to have them all be consistently represented for now.

Reply:
My argument for having this in a separated file is because it makes it easier to see the changes coming in git. I did an extra logic to update the `parser` property.

Reply:
There should be a spec test that snapshots the output of the spec, and that is how we will see changes. You've got that test already so that's great.

In this case, I'd like us to follow the convention of all the other connectors that need to use a hand-written schema because they are beyond the current capabilities of `schema-gen`, which is to have the schema defined as a string [like this](https://github.com/estuary/connectors/blob/a54358b5ef852d5103a421e7ad30da3ef3e7190e/source-sftp/main.go#L441-L552). Another viable option would be to extend `schema-gen` to include the features we need; namely `oneOf` configs, so that we could define the schema as part of the configuration. That would be the best long-term solution, but I don't think its reasonable to require that as a condition of this pr unless you are feeling really ambitious 😄 . But I don't want us to introduce _another_ way of declaring these schemas that is basically still a workaround for `schema-gen`.